libsrc/c64/soft80_cputc.s

```diff
@@ -0,0 +1,463 @@
+; y unmodified
+remcolor:
+
+        ;ldy     #$00            ; is still $00
+
+        ; if the textcolor in vram is equal to the background color, then
+        ; no (visible) character is in the current cell and we can exit
+        ; immediately.
+        lda     (CRAM_PTR),y    ; vram (textcolor)
+        and     #$0f
+        cmp     __bgcolor
+        jeq     @l2b            ; yes, vram==bgcolor
+
+        ; now check if the textcolor in color ram is equal the background color,
+        ; if yes then there is only one (visible) character in the current cell
+        inc     $01
+        lda     (CRAM_PTR),y    ; colram (2nd textcolor)
+        stx     $01             ;$34
+        and     #$0f
+        cmp     __bgcolor
+        beq     @l2s            ; yes, colram==bgcolor
+
+        ; two characters in the current cell, of which one will get removed
+
+        ; vram = colram
+        ;inc     $01
+        ;lda     (CRAM_PTR),y    ; colram
+        ;stx     $01             ;$34
+        ;and     #$0f
+
+        ; move 2nd text color to vram
+        sta     tmp3
+        lda     (CRAM_PTR),y    ; vram
+        and     #$f0
+        ora     tmp3
+        sta     (CRAM_PTR),y    ; vram
+
+        ; colram = bgcolor
+        lda     __bgcolor
+        inc     $01
+        sta     (CRAM_PTR),y    ; colram
+        stx     $01 ;$34
+
+        jmp     @l2b
+
+@l2s:
+        ; colram is bgcolor
+        ; => only one char in cell used
+
+        jsr     soft80_checkchar
+        bcc     @l2b            ; space at current position
+
+        ; vram (textcolor) = bgcolor
+        lda     (CRAM_PTR),y    ; vram
+        and     #$f0
+        ora     __bgcolor
+        sta     (CRAM_PTR),y    ; vram
+@l2b:
+        rts
+
+; put color to cell
+; in: $01 is $34 (RAM under I/O) when entering
+;     y must be $00
+; out: y = $00
+soft80_putcolor:
+
+        ;ldy     #$00            ; is still $00
+
+        lda     (CRAM_PTR),y    ; vram
+        and     #$0f
+        cmp     __bgcolor
+        beq     @l2s            ; vram==bgcolor => first char in cell
+
+        ; vram!=bgcolor => second char in cell
+
+        inc     $01             ;$35
+        lda     (CRAM_PTR),y    ; colram
+        stx     $01             ;$34
+        and     #$0f
+        cmp     __bgcolor
+        bne     @l2s            ; colram!=bgcolor
+
+        ; colram==bgcolor => second char in cell or overwrite 1st char
+
+        jsr     soft80_checkchar
+        bcs     @l2a            ; char at current position => overwrite 1st
+
+        ; colram=vram
+        lda     (CRAM_PTR),y    ; vram
+        inc     $01
+        sta     (CRAM_PTR),y    ; colram
+        stx     $01 ;$34
+
+        ;jmp     @l2a
+
+@l2s:
+        ; colram!=bgcolor => alread 2 chars in cell
+@l2a:
+
+        ; Set color
+        lda     CHARCOLOR
+        sta     (CRAM_PTR),y    ; vram
+
+        rts
+
+        ;.export soft80_checkchar
```

Comment:
I dislike commented-out code without a comment providing the reason for the comment-out. As far as I see this line should just be removed.